pkg/gofr/grpc.go

```diff
@@ -57,3 +58,49 @@ func (g *grpcServer) Shutdown(ctx context.Context) error {
 		return nil
 	})
 }
+
+// RegisterService adds a gRPC service to the GoFr application.
+func (a *App) RegisterService(desc *grpc.ServiceDesc, impl interface{}) {
+	a.container.Logger.Infof("registering GRPC Server: %s", desc.ServiceName)
+	a.grpcServer.server.RegisterService(desc, impl)
+
+	injectContainer(impl, a.container)
+
+	a.grpcRegistered = true
+}
+
+func injectContainer(impl any, c *container.Container) {
```

Comment:
I would have added an interface like this

type containerInjecter interface {
       SetContainer(*container.Container)
}

Then in this code check if impl implements containerInjecter

Then add it to grpc example

Reply:
I did try that approach but then thought that it had to be a lot user-dependent. @vikash thoughts on this suggested approach?

Reply:
One way to do it, without reinventing the wheel, is to use an **existing** dependency injection lib, I think.

I'm afraid about the consequence. Something like, you add a piece of that require a container, maybe for test purpose, and then boom, you get an unexpected behavior

Reply:
@ccoVeille I am afraid that I don't get the consequences exactly. Will testing the gRPC server implementation cause unexpected behaviour?